### PR TITLE
Alerting: Allow to clear datasource selection in panel list

### PR DESCRIPTION
**Pull request by VikaCep**
**What is this feature?**

Adds a "Clear" button that allows to remove the data source selection in the filters section of the alert panel. 
Also improves the text regarding the switch to only show linked alerts.

**Why do we need this feature?**

Previously it wasn't allowed to clear the datasource selection. 

**Who is this feature for?**

All users.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/support-escalations/issues/8250

image

### CODE REVIEW

File: public/app/plugins/panel/alertlist/module.tsx [hunk 1]
```diff
@@ -255,14 +255,19 @@ const unifiedAlertList = new PanelPlugin<UnifiedAlertListOptions>(UnifiedAlertLi
       defaultValue: null,
       editor: function RenderDatasourcePicker(props) {
         return (
-          <DataSourcePicker
-            {...props}
-            type={['prometheus', 'loki', 'grafana']}
-            noDefault
-            current={props.value}
-            onChange={(ds: DataSourceInstanceSettings) => props.onChange(ds.name)}
-            onClear={() => props.onChange(null)}
```
Comment (gillesdemey) [inline public/app/plugins/panel/alertlist/module.tsx:264]: I'm a bit confused, is this perhaps a regression in the new data source picker? Passing `onClear` should make the `Select` component "clearable".

https://github.com/grafana/grafana/blob/ced9fcb94e979f519e113b79da636c6cb71c9683/packages/grafana-runtime/src/components/DataSourcePicker.tsx#L181

Comment (soniaAguilarPeiron) [reply public/app/plugins/panel/alertlist/module.tsx:264]: You're right @gillesdemey , and this regression was introduced [here](https://github.com/grafana/grafana/pull/70609/files#diff-2e980068add69b99836bfe6b81a2d79ce742304bdfea8dbcf689d252adda93d9L3).
 I tested locally with the DataSourcePicker we were using before and the clear was working correctly. 
 
Screenshot 2023-11-10 at 09 12 59

Comment (VikaCep) [reply public/app/plugins/panel/alertlist/module.tsx:264]: Right, the new datasource picker doesn't care about the `onClear` function https://github.com/grafana/grafana/blob/main/public/app/features/datasources/components/picker/DataSourceDropdown.tsx#L197

File: public/app/plugins/panel/alertlist/module.tsx [hunk 2]
```diff
@@ -255,14 +255,19 @@ const unifiedAlertList = new PanelPlugin<UnifiedAlertListOptions>(UnifiedAlertLi
...
-          />
+          <Stack gap={1}>
+            <DataSourcePicker
+              {...props}
+              type={['prometheus', 'loki', 'grafana']}
+              noDefault
+              current={props.value}
+              onChange={(ds: DataSourceInstanceSettings) => props.onChange(ds.name)}
+              onClear={() => props.onChange(null)}
```
Comment (soniaAguilarPeiron) [inline public/app/plugins/panel/alertlist/module.tsx:265]: if this `onClear` does not work, we can remove it?

Comment (VikaCep) [reply public/app/plugins/panel/alertlist/module.tsx]: Removed it :)

**Review by gillesdemey — APPROVED** [review]
I'm going to approve the PR as-is for now – it does solve the immediate problem and we can go back and update once we've landed changes in the data source picker to allow clearing the selection.

**Pull request merged**